docs/how-to/use-nhwc-batchnorm-in-pytorch.rst

```diff
@@ -0,0 +1,181 @@
+
+(*) MIOpen is used with ROCm 6.4 and later. Otherwise, the native backend is used.
+
+
+Disabling MIOpen for Batchnorm in PyTorch
+=================================================
+
+In some situations, you might not want to use MIOpen as the backend for Batchnorm operations.
+To disable the use of MIOpen with Batchnorm, add this code to your application.
+
+.. code:: python
+
+   inp = torch.randn(size, requires_grad=True)
+   grad = torch.randn(size, requires_grad=False)
+   mod = nn.BatchNorm2d(inp.size(1), device="cuda")
+
+   with torch.backends.cudnn.flags(enabled=False): # this line disables MIOpen for the two lines below, native batchnorm will be used
+
+      out = mod(inp)
+      out.backward(grad)
+
+Verifying NHWC Batchnorm use with MIOpen
+=================================================
+
+For some operations, it can be difficult to determine the backend and memory format that were used.
+To verify whether MIOpen is being used and whether the memory format is NHWC or NCHW, run your program
+with the following environment variable:
+
+.. code:: shell
+
+   MIOPEN_ENABLE_LOGGING_CMD=1
+
+Here is an example command:
+
+.. code:: shell
+
+   MIOPEN_ENABLE_LOGGING_CMD=1 python test_nn.py -v -k test_batchnorm_cudnn_nhwc
+
+The output might look like this:
+
+.. code:: shell
+
+   MIOpen(HIP): Command [LogCmdBNorm] ./bin/MIOpenDriver bnorm -n 4 -c 8 -H 2 -W 2 -m 1 --forw 1 -b 0 -r 1 -s 1 --layout NHWC
+   MIOpen(HIP): Command [LogCmdBNorm] ./bin/MIOpenDriver bnorm -n 4 -c 8 -H 2 -W 2 -m 1 --forw 0 -b 1 -s 1 --layout NHWC
+   MIOpen(HIP): Command [LogCmdBNorm] ./bin/MIOpenDriver bnorm -n 4 -c 8 -H 2 -W 2 -m 1 --forw 1 -b 0 -r 1 -s 1 --layout NCHW
+   MIOpen(HIP): Command [LogCmdBNorm] ./bin/MIOpenDriver bnorm -n 4 -c 8 -H 2 -W 2 -m 1 --forw 0 -b 1 -s 1 --layout NCHW
+   MIOpen(HIP): Command [LogCmdBNorm] ./bin/MIOpenDriver bnorm -n 2 -c 8 -H 8 -W 1 -m 1 --forw 1 -b 0 -r 1 -s 1 --layout NHWC
+   MIOpen(HIP): Command [LogCmdBNorm] ./bin/MIOpenDriver bnorm -n 2 -c 8 -H 8 -W 1 -m 1 --forw 0 -b 1 -s 1 --layout NHWC
+   MIOpen(HIP): Command [LogCmdBNorm] ./bin/MIOpenDriver bnorm -n 2 -c 8 -H 8 -W 1 -m 1 --forw 1 -b 0 -r 1 -s 1 --layout NCHW
+   MIOpen(HIP): Command [LogCmdBNorm] ./bin/MIOpenDriver bnorm -n 2 -c 8 -H 8 -W 1 -m 1 --forw 0 -b 1 -s 1 --layout NCHW
+
+Each line corresponds to a different command or operation.
+The ``./bin/MIOpenDriver`` string indicates that MIOpen was used for the operation.
+The ``--layout`` parameter shows whether NHWC or NCHW was used, for example, ``--layout NHWC`` means the
+NHWC memory format was used.
+
+Running Batchnorm tests
+=======================
+
+Several test suites are available for Batchnorm. To test Batchnorm training using both NHWC and NCHW,
+run the following command:
+
+.. code:: shell
+
+   python test_nn.py -v -k test_batchnorm_train
+
+To test Batchnorm inference using both memory formats, use this command:
+
+.. code:: shell
+
+   python test_nn.py -v -k test_batchnorm_inference
```

Comment:
@dnikolaev-amd Is the test name pattern changed after introduction of 3D batchnorms support in our release branches?